prometheus/exporter/gpu_exporter.py

```diff
@@ -54,9 +54,12 @@ def collect_gpu_info():
     try:
         logger.info("call nvidia-smi to get gpu metrics")
 
-        smi_output = utils.check_output(["nvidia-smi", "-q", "-x"])
-
-        return parse_smi_xml_result(smi_output)
+        nvidia_smi_exists = len(utils.check_output(["which", "nvidia-smi"])) > 0
```

Comment:
catch OSError directly like [this](https://stackoverflow.com/a/11210185) so you do not have to call `which`

Reply:
refactored